### PR TITLE
[libzypp] Do not disable libproxy use. JB#56759

### PR DESCRIPTION
**Pull request by martyone**

### CODE REVIEW

**Comment by martyone** [comment]
Despite this `-DDISABLE_LIBPROXY=TRUE`, which was added during upgrade to 17.24.2 with commit 7be7f21ba2117aeaa098668936b08ae1cdcb50d5, there is also `BuildRequires:  pkgconfig(libproxy-1.0)` left from past. So I guess the option appeared as new during that upgrade, which was wrongly perceived as that the feature to use libproxy was new and so it was disabled as unneeded, missing the fact that it was actually used before.

File: rpm/libzypp.spec [hunk 1]
```diff
@@ -94,7 +94,6 @@ export CXXFLAGS="%{optflags}"
       -DCMAKE_BUILD_TYPE=Release \
       -DCMAKE_SKIP_RPATH=1 \
       -DUSE_TRANSLATION_SET=${TRANSLATION_SET:-zypp} \
-      -DDISABLE_LIBPROXY=TRUE \
```
Comment (saukko) [inline rpm/libzypp.spec:97]: If this would be =FALSE would it enforce this always and fail in case someone removes libproxy?

Comment (Thaodan) [reply rpm/libzypp.spec:97]: Not it doesn't:
https://github.com/sailfishos-mirror/libzypp/blob/43bd2bf563be01d429e7930019da7682ef430c1c/CMakeLists.txt#L242

**Comment by Thaodan** [comment]
> Despite this `-DDISABLE_LIBPROXY=TRUE`, which was added during upgrade to 17.24.2 with commit [7be7f21](https://github.com/sailfishos/libzypp/commit/7be7f21ba2117aeaa098668936b08ae1cdcb50d5), there is also `BuildRequires: pkgconfig(libproxy-1.0)` left from past. So I guess the option appeared as new during that upgrade, which was wrongly perceived as that the feature to use libproxy was new and so it was disabled as unneeded, missing the fact that it was actually used before.

Please add this to the commit message as context.

File: rpm/libzypp.spec (hunk 1 above)
Comment (martyone) [reply rpm/libzypp.spec:97]: Unfortunately it doesn't work that way

**Comment by martyone** [comment]
I found that it would be more clear to have a separate bug for this, so replaced with PR #3 .

@Thaodan I extended the commit message there.

**Pull request closed without merging**